### PR TITLE
Daffodil 2899

### PR DESCRIPTION
**Pull request by parkera101093**
SDW now properly gives VariableRead WarnID for case VariableRead in setVariable funcion

### CODE REVIEW

File: daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/VariableMap1.scala [hunk 1]
```diff
@@ -455,14 +455,14 @@ class VariableMap private (
          * the way we evaluate escapeSchemes it could lead us to setting the
          * variable read too early */
         pstate.SDW(
-          WarnID.VariableSet,
+          WarnID.VariableRead,
           "Cannot set variable %s after reading the default value. State was: %s. Existing value: %s",
           variable.rd.globalQName,
-          VariableSet,
+          VariableRead,
```
Comment (stevedlawrence) [inline daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/VariableMap1.scala:461]: Is line 461the only line that needs to be changed?

This warning is about setting a variable, so I think the WarnID should probably still be `VariableSet`.

And the variable state needs to be set to "VariableSet" below since we are changing the variable state from Read to Set. I think all that needs to change is the warning from "State was VariableSet" to "State was VariableRead", which this line does.

Also, please rebase/squash your changes into a single commit and include the bug number in the commit message. The [Code Contributor Workflow](https://cwiki.apache.org/confluence/display/DAFFODIL/Code+Contributor+Workflow) has commands and an example commit message if you're unsure.

**Review by stevedlawrence — CHANGES_REQUESTED** [review]

File: daffodil-propgen/src/main/resources/org/apache/daffodil/xsd/dafext.xsd [hunk 2]
```diff
@@ -735,6 +735,7 @@
           <xs:enumeration value="unsupportedAttributeFormDefault" />
           <xs:enumeration value="unsupportedAttributeSchemaLocation" />
           <xs:enumeration value="variableSet" />
+          <xs:enumeration value="variableRead" />
```
Comment (stevedlawrence) [inline daffodil-propgen/src/main/resources/org/apache/daffodil/xsd/dafext.xsd:738]: Can this be removed? I don't think this new warn ID is used or neeeded?

File: daffodil-test/src/test/resources/org/apache/daffodil/section07/variables/variables.tdml [hunk 3]
```diff
@@ -24,7 +24,7 @@
   xmlns:fn="http://www.w3.org/2005/xpath-functions"
   xmlns:ex="http://example.com"
   defaultRoundTrip="true">
-  
+
```
Comment (stevedlawrence) [inline daffodil-test/src/test/resources/org/apache/daffodil/section07/variables/variables.tdml:27]: In general, I find it's best to avoid these kinds of unrelated whitespace changes to make reviewing the actual changes easier. Sometimes this requires tweaks to IDE settings to disable this. We probably should fix the whitespace in our TDML files at some point, but we should that to everything at once like we did when we standardized our scala formatting.

File: daffodil-test/src/test/resources/org/apache/daffodil/section07/variables/variables.tdml [hunk 4]
```diff
@@ -2217,6 +2217,9 @@
         </root>
       </tdml:dfdlInfoset>
     </tdml:infoset>
+    <tdml:warnings>
+      <tdml:warning>State was: VariableRead</tdml:warning>
```
Comment (stevedlawrence) [inline daffodil-test/src/test/resources/org/apache/daffodil/section07/variables/variables.tdml:2221]: This should be moved up into the other `<tdml:warnings>` tag, makes it easier if all the warnings are in the same place, and there really isn't anything gained by having them separate--I didn't even realize we supported that.

File: daffodil-test/src/test/resources/org/apache/daffodil/section07/variables/variables.tdml [hunk 5]
```diff
@@ -2646,6 +2649,9 @@
...
+    </tdml:warnings>
```
Comment (stevedlawrence) [inline daffodil-test/src/test/resources/org/apache/daffodil/section07/variables/variables.tdml:2654]: I think this `<tdml:warnings>` can be removed. Although this test might result in that warning, it really isn't specifically about that warning--it's really just making sure some escape character thing works. Whether or not there is a warning doesn't really matter to this tests.

The above tests that you've modified is specifically about this warning and making sure it's reasonable and correct.

**Review by stevedlawrence — COMMENTED** [review]
Looks good, just a couple minor this to clean up.

Also, the commit message says it corrects the WarnID, but that hasn't changed. All that has changed is the warning message to include the correct state.

**Review by stevedlawrence — APPROVED** [review]
+1

**Review by mbeckerle — APPROVED** [review]
+1 nothing to add beyond Steve's comments.
